lib/puppet/module.rb

```diff
@@ -180,7 +180,7 @@ def match_manifests(rest)
 
     # (#4220) Always ensure init.pp in case class is defined there.
     init_manifest = manifest("init.pp")
-    if !init_manifest.nil? && !searched_manifests.include?(init_manifest)
+    if !init_manifest.nil? && !searched_manifests.include?(Dir.glob(init_manifest).first)
```

Comment:
So this would search for the file on disk again to get the path in the same form.